coder_sniffer/Drupal/Sniffs/Commenting/TodoCommentSniff.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Parses and verifies comment language.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     http://pear.php.net/package/PHP_CodeSniffer
+ */
+
+namespace Drupal\Sniffs\Commenting;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+/**
+ * Parses and verifies that comments use the correct @todo format.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     http://pear.php.net/package/PHP_CodeSniffer
+ */
+class TodoCommentSniff implements Sniff
+{
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array<int|string>
+     */
+    public function register()
+    {
+        return [
+            T_COMMENT,
+            T_DOC_COMMENT_TAG,
+        ];
+
+    }//end register()
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+        // Use the comment text by default.
+        $comment = $tokens[$stackPtr]['content'];
+        // Use the next line for multi-line comments.
+        if ($tokens[$stackPtr]['code'] === T_DOC_COMMENT_TAG) {
```

Comment:
I used `$tokens[$stackPtr]['type']` here. I don't know which is preferred.

Reply:
`$tokens[$stackPtr]['code'] === T_DOC_COMMENT_TAG` will never be true. It should be either 
````
$tokens[$stackPtr]['type'] === T_DOC_COMMENT_TAG
````
or
````
$tokens[$stackPtr]['code'] === 'PHPCS_T_DOC_COMMENT_TAG'
````
The 'type' check I think is preferrable, because they are always readable. Sometimes 'code' is a number, other times it is a string as above, whereas checking 'type' can always use the constants and will be consistent.

Reply:
Other sniffs seem to use the `T_DOC_*` constants, which I assume is preferable to the `'PHPCS_T_DOC_COMMENT_TAG'` string? Especially since that string is simple the value of that constant.

```
// Tokens used for parsing doc blocks.
define('T_DOC_COMMENT_STAR', 'PHPCS_T_DOC_COMMENT_STAR');
define('T_DOC_COMMENT_WHITESPACE', 'PHPCS_T_DOC_COMMENT_WHITESPACE');
define('T_DOC_COMMENT_TAG', 'PHPCS_T_DOC_COMMENT_TAG');
define('T_DOC_COMMENT_OPEN_TAG', 'PHPCS_T_DOC_COMMENT_OPEN_TAG');
define('T_DOC_COMMENT_CLOSE_TAG', 'PHPCS_T_DOC_COMMENT_CLOSE_TAG');
define('T_DOC_COMMENT_STRING', 'PHPCS_T_DOC_COMMENT_STRING');
```

Thoughts?

Reply:
Yes the `tokens[$stackPtr]['type'] === T_DOC_*` way is preferrable, and this is what you've done.